### PR TITLE
update bastion machine type to variable in AWS HA terraform

### PR DESCRIPTION
**Pull request by stevenGravy**
Sets to use a variable instance type instead of hard-coded. This currently causes problems if you specify a x86 AMI since that's not compatible with arm types.

changelog: Allow specifying the instance type of AWS HA Terraform bastion instance

### CODE REVIEW

**Review by greedy52 — APPROVED** [review]

File: examples/aws/terraform/ha-autoscale-cluster/vars.tf [hunk 1]
```diff
@@ -38,6 +38,12 @@ variable "node_instance_type" {
   default = "t4g.medium"
 }
 
+// Instance type used for bastion server
+variable "bastion_instance_type" {
+  type    = string
+  default = "t4g.medium"
```
[the remark this answers is not in the document]
Comment (stevenGravy) [reply examples/aws/terraform/ha-autoscale-cluster/vars.tf:44]: Yes, that was a change here already.

**Review by hugoShaka — COMMENTED** [review]
So we broke the TF code by default for every x86 user 6 moths ago, but did not catch the issue until now because we default to ARM?

The new TF variable should be added to the Makefile and the README.

**Comment by hugoShaka** [comment]
Also, this PR is changelog worthy.

**Comment by stevenGravy** [comment]
> So we broke the TF code by default for every x86 user 6 moths ago, but did not catch the issue until now because we default to ARM?
> 
> The new TF variable should be added to the Makefile and the README.

We don't put the other instance types (auth, proxy,...) though in the `Makefile` and `README`.  Should they all be there then?

**Comment by stevenGravy** [comment]
> Also, this PR is changelog worthy.

Thanks, changelog updated.

**Comment by hugoShaka** [comment]
> We don't put the other instance types (auth, proxy,...) though in the Makefile and README. Should they all be there then?

The Makefile and README should contain every variable and mention if it's required or if we'll pick the default for the user. It's very likely that we missed adding variables there in the past. If you find missing vars, please add them to the reference 🙏

File: examples/aws/terraform/ha-autoscale-cluster/vars.tf (hunk 1 above)
Comment (fheinecke) [inline examples/aws/terraform/ha-autoscale-cluster/vars.tf:44]: In the interest of having a more "friendly" experience for new users, maybe we could default this to null, then conditionally set the instance type deployed based off of the AMI architecture from `data.aws_ami.base`?

**Comment by stevenGravy** [comment]
> > We don't put the other instance types (auth, proxy,...) though in the Makefile and README. Should they all be there then?
> 
> The Makefile and README should contain every variable and mention if it's required or if we'll pick the default for the user. It's very likely that we missed adding variables there in the past. If you find missing vars, please add them to the reference 🙏

thanks @hugoShaka, please see update.

**Review by hugoShaka — APPROVED** [review]
LGTM, but I'd like @webvictim 's approval before merging.

File: examples/aws/terraform/ha-autoscale-cluster/Makefile [hunk 2]
```diff
@@ -21,6 +21,23 @@ TF_VAR_license_path ?=
 # FIPS 140-2 images are also available for Enterprise customers, look for '-fips' on the end of the AMI's name
 TF_VAR_ami_name ?=
 
+
+# Instance types used for authentication servers auto scale group
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/Makefile:25]:
```suggestion
# Instance types used for authentication server auto scaling group
```

File: examples/aws/terraform/ha-autoscale-cluster/Makefile [hunk 3]
```diff
@@ -21,6 +21,23 @@ TF_VAR_license_path ?=
...
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/Makefile:26]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/Makefile [hunk 4]
```diff
@@ -21,6 +21,23 @@ TF_VAR_license_path ?=
...
+TF_VAR_auth_instance_type ?= m7g.large
+
+# Instance types used for proxy auto scale groups
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/Makefile:30]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/Makefile [hunk 5]
```diff
@@ -21,6 +21,23 @@ TF_VAR_license_path ?=
...
+TF_VAR_proxy_instance_type =? m7g.large
+
+# Instance types used for teleport nodes auto scale groups
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/Makefile:33]:
```suggestion
# Instance types used for Teleport node auto scaling group
```

File: examples/aws/terraform/ha-autoscale-cluster/Makefile [hunk 6]
```diff
@@ -21,6 +21,23 @@ TF_VAR_license_path ?=
...
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/Makefile:34]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/Makefile [hunk 7]
```diff
@@ -21,6 +21,23 @@ TF_VAR_license_path ?=
...
+# Instance types used for proxy auto scale groups
+# This should match to the AMI instance architecture type, Arm or x86
+TF_VAR_proxy_instance_type =? m7g.large
+
+# Instance types used for teleport nodes auto scale groups
+# This should match to the AMI instance architecture type, Arm or x86 
+TF_VAR_node_instance_type ?= t4g.medium
+
+# Instance type used for bastion server
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/Makefile:38]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/Makefile [hunk 8]
```diff
@@ -21,6 +21,23 @@ TF_VAR_license_path ?=
...
+TF_VAR_proxy_instance_type =? m7g.large
+
+# Instance types used for teleport nodes auto scale groups
+# This should match to the AMI instance architecture type, Arm or x86 
+TF_VAR_node_instance_type ?= t4g.medium
+
+# Instance type used for bastion server
+# This should match to the AMI instance architecture type, Arm or x86 
+TF_VAR_bastion_instance_type ?=  t4g.medium
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/Makefile:39]:
```suggestion
TF_VAR_bastion_instance_type ?= t4g.medium
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 9]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
 # FIPS 140-2 images are also available for Enterprise customers, look for '-fips' on the end of the AMI's name
 export TF_VAR_ami_name="teleport-ent-16.4.2-arm64"
 
+# Instance types used for authentication servers auto scale group
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:51]:
```suggestion
# Instance types used for authentication server auto scaling group
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 10]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
...
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:52]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 11]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
...
+TF_VAR_auth_instance_type ?= m7g.large
+
+# Instance types used for proxy auto scale groups
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:55]:
```suggestion
# Instance types used for proxy auto scaling group
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 12]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
...
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:56]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 13]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
...
+TF_VAR_proxy_instance_type =? m7g.large
+
+# Instance types used for teleport nodes auto scale groups
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:59]:
```suggestion
# Instance types used for Teleport node auto scaling group
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 14]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
...
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:60]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 15]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
...
+TF_VAR_node_instance_type ?= t4g.medium
+
+# Instance type used for bastion server
+# This should match to the AMI instance architecture type, Arm or x86
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:64]:
```suggestion
# This should match to the AMI instance architecture type, ARM or x86
```

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 16]
```diff
@@ -48,6 +48,22 @@ export TF_VAR_cluster_name="teleport.example.com"
...
+TF_VAR_bastion_instance_type ?=  t4g.medium
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:65]:
```suggestion
TF_VAR_bastion_instance_type ?= t4g.medium
```

File: examples/aws/terraform/ha-autoscale-cluster/vars.tf (hunk 1 above)
Comment (webvictim) [reply examples/aws/terraform/ha-autoscale-cluster/vars.tf:44]: I really like this idea, but think it's probably better implemented in a future PR as it'll require a fair bit more testing.

**Review by webvictim — APPROVED** [review]
LGTM, just grammar/formatting nits

**Pull request merged**